### PR TITLE
Enhance bm files download and add dispatcher to bm flow

### PR DESCRIPTION
**Pull request by meiravgri**
1. Add workflow dispatcher that allows triggering the benchmarks manually
2. parallelize benchmark files download. Downloading all the benchmarks files takes ~5 min instead of more than 20 min.

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -10,23 +10,12 @@ on:
...
-          - bm-batch-iter-fp32-single
-          - bm-batch-iter-fp32-multi
-          - bm-batch-iter-fp64-single
-          - bm-batch-iter-fp64-multi
-          - bm-batch-iter-bf16-single
-          - bm-batch-iter-bf16-multi
-          - bm-batch-iter-fp16-single
-          - bm-batch-iter-fp16-multi
           - bm-updated-fp32-single
           - bm-spaces
```
Comment (GuyAv46) [inline .github/workflows/benchmark.yml:20]: why not include the full list?

Comment (meiravgri) [reply .github/workflows/benchmark.yml:20]: I read that it is limited to 10 but then read again and they are talking about the input list

Comment (GuyAv46) [reply .github/workflows/benchmark.yml:31]: Really? Good to know

File: tests/benchmark/bm_files.sh [hunk 2]
```diff
@@ -32,4 +32,4 @@ elif [ "$BM_TYPE" = "bm-updated-fp32-single" ]; then
     file_name="updated"
 fi
 
-wget --no-check-certificate -q -i tests/benchmark/data/hnsw_indices/hnsw_indices_$file_name.txt -P tests/benchmark/data
```
Comment (GuyAv46) [inline tests/benchmark/bm_files.sh:35]: Intentionally removing the `-q`?

Comment (meiravgri) [reply tests/benchmark/bm_files.sh:35]: Yes, it doesn't affect the time and might be useful if something is wrong

**Review by GuyAv46 — DISMISSED** [review]

**Review by GuyAv46 — APPROVED** [review]

**Pull request merged**